openmetadata-ui/src/main/resources/ui/src/components/Services/Services.tsx

```diff
@@ -0,0 +1,264 @@
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { Card } from 'antd';
+import { isNil } from 'lodash';
+import { EntityReference } from 'Models';
+import React, { Fragment } from 'react';
+import { Link, useHistory } from 'react-router-dom';
+import { useAuthContext } from '../../authentication/auth-provider/AuthProvider';
+import {
+  getServiceDetailsPath,
+  PAGE_SIZE,
+  TITLE_FOR_NON_ADMIN_ACTION,
+} from '../../constants/constants';
+import {
+  NoDataFoundPlaceHolder,
+  servicesDisplayName,
+} from '../../constants/services.const';
+import { ServiceCategory } from '../../enums/service.enum';
+import { DashboardService } from '../../generated/entity/services/dashboardService';
+import { MessagingService } from '../../generated/entity/services/messagingService';
+import { MlmodelService } from '../../generated/entity/services/mlmodelService';
+import { PipelineService } from '../../generated/entity/services/pipelineService';
+import { Paging } from '../../generated/type/paging';
+import { useAuth } from '../../hooks/authHooks';
+import { DataService } from '../../interface/service.interface';
+import { getEntityName, getServiceLogo } from '../../utils/CommonUtils';
+import { getDashboardURL } from '../../utils/DashboardServiceUtils';
+import { getBrokers } from '../../utils/MessagingServiceUtils';
+import { getAddServicePath } from '../../utils/RouterUtils';
+import { Button } from '../buttons/Button/Button';
+import NextPrevious from '../common/next-previous/NextPrevious';
+import NonAdminAction from '../common/non-admin-action/NonAdminAction';
+import RichTextEditorPreviewer from '../common/rich-text-editor/RichTextEditorPreviewer';
+import { leftPanelAntCardStyle } from '../containers/PageLayout';
+
+interface ServicesProps {
+  serviceData: DataService[];
+  serviceName: string;
+  paging: Paging;
+  currentPage: number;
+  onPageChange: (cursorType: string | number, activePage?: number) => void;
+}
+
+const Services = ({
+  serviceData,
+  serviceName,
+  paging,
+  currentPage,
+  onPageChange,
+}: ServicesProps) => {
+  const { isAdminUser } = useAuth();
+  const { isAuthDisabled } = useAuthContext();
+  const history = useHistory();
+  const goToAddService = () => {
+    history.push(getAddServicePath(serviceName));
+  };
+  const getOptionalFields = (service: DataService): JSX.Element => {
+    switch (serviceName) {
+      case ServiceCategory.MESSAGING_SERVICES: {
+        const messagingService = service as MessagingService;
+
+        return (
+          <div className="tw-mb-1 tw-truncate" data-testid="additional-field">
+            <label className="tw-mb-0">Brokers:</label>
+            <span
```

Comment:
Can this be part of utils? Or it's specific to settings page?

Reply:
This is specific to the service page